### DISCUSSION TITLE
List all commits before a given commit with GraphQL API

### DISCUSSION THREAD

**Discussion by caithagoras** [community/community#24561] [Apps, API and Webhooks]
<p>I’m trying to list all commit on a branch between the latest commit and a given commit.</p>
<p>It seems that the history(before) is not working as intended. The query below gives me the first 100 commit on that branch without respecting the before cursor I specified. history(after) seems to be working just fine. Am I misunderstanding something here?</p>
<pre><code>{
  repository(owner: "prestodb", name: "presto") {
        ref(qualifiedName: "release-0.235") {
            target {
                ... on Commit {
                    history(before: "f9f9d83c84dbb35fc49e5b4c70326024d8ff3d1e 0") {
                        pageInfo {
                            hasNextPage
                            endCursor
                        }
                        edges {
                            node {
                                oid
                                message
                                # author {
                                # name
                                # }
                            }
                        }
                    }
                }
            }
        }
    }
}
</code></pre>

**Comment by KnowYourselves** [comment, marked as answer]
<p>Hello <a class="mention" href="/caithagoras">@caithagoras</a>!</p>
<p>The problem with your query is that you’re asking what’s before the first cursor. I know that because cursors end with their index and apparently, if you ask what’s before the first index, you get all of them, but since the max amount of nodes per request is 100, its giving you 100 commits.</p>
